internal/data/resource_repository.go

```diff
@@ -241,3 +248,34 @@ func (r *resourceRepository) GetDB() *gorm.DB {
 func (r *resourceRepository) GetTransactionManager() usecase.TransactionManager {
 	return r.transactionManager
 }
+
+func (r *resourceRepository) FindVersionedRepresentationsByVersion(tx *gorm.DB, key bizmodel.ReporterResourceKey, currentVersion uint) ([]RepresentationsByVersion, error) {
```

Comment:
Add a TODO here to say this is focused on getting common version for now but will need to be expanded to get reporter representation version as well later